WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsUnknownViewController.swift

```diff
@@ -39,6 +45,57 @@ final class CardReaderSettingsUnknownViewController: UIViewController, CardReade
         configureSections()
         configureTable()
     }
+
+    /// We're disappearing... stop listening to viewmodel changes to avoid a reference loop
+    /// and close any modal
+    ///
+    override func viewWillDisappear(_ animated: Bool) {
+        viewModel?.didUpdate = nil
+        alert?.dismiss(animated: false, completion: nil)
+        super.viewWillDisappear(animated)
+    }
+}
+
+// MARK: - View Updates
+//
+private extension CardReaderSettingsUnknownViewController {
+    func onViewModelDidUpdate() {
+        updateModal()
+        updateTable()
+    }
+
+    func updateModal() {
+        guard let viewModel = viewModel else {
+            return
+        }
+
+        /// Dismiss any pre-existing modal
+        alert?.dismiss(animated: true, completion: nil)
+
+        if viewModel.discoveryState == .searching {
+            showSearchingModal()
+        }
+    }
+
+    func updateTable() {
+        tableView.reloadData()
+    }
+
+    func showSearchingModal() {
+        // TODO - replace with new, richer modals once they stabilize
```

Comment:
I think the new modals should be stable enough for this? Is there anything in particular missing? 

Anyway, would you mind logging a ticket for this (or pointing to an existing one?)